### PR TITLE
Support: Worker Pool Taints

### LINKED ISSUE #2696: Add support for managing Kubernetes/OpenShift worker taints
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

It's possible to set the taints for workers using the `ibmcloud oc` tool, but it would be much more useful to be able to manage these using Terraform.

Looking at the API that's vendored under `common/` it looks like there isn't an API to _read_ the current taints applied to a worker pool, only set.

### New or Affected Resource(s)

* ibm_container_worker_pool (resource & data source)
* ibm_container_vpc_worker_pool (resource & data source)
* ibm_container_cluster - (just data source to avoid stopping the default pool from running system components)
* ibm_container_vpc_cluster - (just data source to avoid stopping the default pool from running system components)

### Potential Terraform Configuration

```hcl
resource "ibm_container_vpc_worker_pool" "worker_pool" {
  cluster           = ibm_container_vpc_cluster.cluster.name
  worker_pool_name  = "name"
  # ...

  taints = {
    "env": "prod:NoSchedule"
  }
}
```

alternatively:
```hcl
resource "ibm_container_vpc_worker_pool" "worker_pool" {
  cluster           = ibm_container_vpc_cluster.cluster.name
  worker_pool_name  = "name"
  # ...

  taint {
    key = "env"
    value = "prod"
    effect = "NoSchedule"
  }
}
```

### References

* #0000

### PR DESCRIPTION
**Pull request by kavya498**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #2696  #1646  #624 #2862

### CODE REVIEW

File: ibm/provider.go [hunk 1]
```diff
@@ -670,6 +670,10 @@ func Validator() ValidatorDict {
 				"ibm_cis_waf_rule":                      resourceIBMCISWAFRuleValidator(),
 				"ibm_cis_certificate_order":             resourceIBMCISCertificateOrderValidator(),
 				"ibm_cis_filter":                        resourceIBMCISFilterValidator(),
+				"ibm_container_cluster":                 resourceIBMContainerClusterValidator(),
+				"ibm_container_worker_pool":             resourceContainerWorkerPoolTaintsValidator(),
```
Comment (hkantare) [inline ibm/provider.go:674]: remove Tainsts (resourceContainerWorkerPoolValidator, resourceContainerVPCWorkerPoolValidator)

File: ibm/resource_ibm_container_cluster.go [hunk 2]
```diff
@@ -1042,6 +1076,41 @@ func resourceIBMContainerClusterUpdate(d *schema.ResourceData, meta interface{})
...
+			}
+			ClusterClient, err := meta.(ClientSession).VpcContainerAPI()
+			if err != nil {
+				return err
+			}
+			err = ClusterClient.WorkerPools().UpdateWorkerPoolTaints(taintParam, targetEnv)
+			if err != nil {
+				return fmt.Errorf("[ERROR] Error updating the taints: %s", err)
+			}
+		} else {
```
Comment (hkantare) [inline ibm/resource_ibm_container_cluster.go:1109]: Do we need else block here?

File: website/docs/r/container_cluster.html.markdown [hunk 3]
```diff
@@ -226,6 +226,13 @@ Review the argument references that you can specify for your resource.
 - `subnet_id` - (Optional, String) The ID of an existing subnet that you want to use for your worker nodes. To find existing subnets, run `ibmcloud ks subnets`.
 - `service_subnet`-  (Optional, Forces new resource, String) Specify a custom subnet CIDR to provide private IP addresses for services. The subnet should be at least `/24` or more. For more information, refer to [Subnet service](https://cloud.ibm.com/docs/containers?topic=containers-cli-plugin-kubernetes-service-cli#service-subnet).
 - `tags` - (Optional, Array of string)  A list of tags that you want to add to your cluster. Tags can help find a cluster more quickly.  **Note**: For users on account to add tags to a resource, they must be assigned the appropriate [permissions](https://cloud.ibm.com/docs/resources?topic=resources-access).
+- `taints` - (Optional, List) A nested block describes the taints of this worker pool.
```
Comment (hkantare) [inline website/docs/r/container_cluster.html.markdown:229]: change description to `Set and remove Kubernetes taints for all worker nodes in a worker pool.` in all docs

File: ibm/resource_ibm_container_cluster.go [hunk 4]
```diff
@@ -580,6 +607,7 @@ func resourceIBMContainerCluster() *schema.Resource {
 }
 
 func resourceIBMContainerClusterValidator() *ResourceValidator {
+	tainteffects := "NoSchedule, PreferNoSchedule, and NoExecute"
```
Comment (PKehnel) [inline ibm/resource_ibm_container_cluster.go:610]: `tainteffects := "NoSchedule, PreferNoSchedule, NoExecute"`

**Pull request merged**